File: benchmark/go/task6_fileio.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func countWords(inputFile, outputFile string) error {
	file, err := os.Open(inputFile)
	if err != nil {
		return err
	}
	defer file.Close()

	counts := make(map[string]int)
	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanWords)
	for scanner.Scan() {
		counts[scanner.Text()]++
	}

	out, err := os.Create(outputFile)
	if err != nil {
		return err
	}
	defer out.Close()

	for word, count := range counts {
		fmt.Fprintf(out, "%s: %d\n", word, count)
	}
	return nil
}

File: benchmark/perl/task6_fileio.pl
sub count_words {
    my ($input_file, $output_file) = @_;
    my %counts;

    open my $in, '<', $input_file or die $!;
    while (<$in>) {
        $counts{$_}++ for split;
    }
    close $in;

    open my $out, '>', $output_file or die $!;
    print $out "$_: $counts{$_}\n" for keys %counts;
    close $out;
}
